### PR TITLE
Imports from collections

### PR DESCRIPTION
**Pull request by keriksson-rosenqvist**
Adjust imports that have been deprecated in py3.10 while retaining compatibility with py3.8.

### CODE REVIEW

**Comment by minsu-seo-oqc** [comment]
I have read the Contributor License Agreement and I hereby accept the Terms.

**Review by minsu-seo-oqc — APPROVED** [review]

**Review by hamidelmaazouz — APPROVED** [review]

File: src/QAT/qat/purr/compiler/builders.py [hunk 1]
```diff
@@ -2,7 +2,12 @@
 # Copyright (c) 2023 Oxford Quantum Circuits Ltd
 import itertools
 import math
-from collections import defaultdict
+import sys
+
+if sys.version_info < (3, 10):
```
Comment (jfriel-oqc) [inline src/QAT/qat/purr/compiler/builders.py:7]: why do we need this?

```
kristianb@jamie-xps-15-9520:~/Documents/code/scrapyard/pyt39_tes$ poetry run python
Python 3.8.10 (default, Nov 22 2023, 10:22:35) 
[GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from collections.abc import Iterable
>>> 
```

**Review by jfriel-oqc — CHANGES_REQUESTED** [review]

File: src/QAT/qat/purr/compiler/builders.py (hunk 1 above)
Comment (keriksson-rosenqvist) [reply src/QAT/qat/purr/compiler/builders.py:7]: Interesting. It fails import that way on my machine and in CI. :thinking:

Comment (keriksson-rosenqvist) [reply src/QAT/qat/purr/compiler/builders.py:7]: As seen here.
https://github.com/oqc-community/qat/actions/runs/9743002674/job/26885510222#step:5:334

**Comment by keriksson-rosenqvist** [comment]
The necessary adjustments are being handled in other PRs. Dropping this in support of them.

**Pull request closed without merging**